scripts/addons.py

```diff
@@ -0,0 +1,249 @@
+# -*- coding: utf-8 -*-
+# Copyright 2015 Elico Corp
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+import os
+import re
+from os import remove
+from shutil import move
+from urlparse import urlparse
+from subprocess import call
+from subprocess import check_output
+
+EXTRA_ADDONS_PATH = '/mnt/data/additional_addons/'
+OLD_ODOO_CONF = '/opt/odoo/etc/odoo.conf.old'
+ODOO_CONF = '/opt/odoo/etc/odoo.conf'
+ADDONS_PATH = ['/opt/odoo/sources/odoo/addons']
+DEFAULT_SCHEME = 'https://'
+DEFAULT_GIT_REPO_HOSTING_SERVICE = 'github.com'
+DEFAULT_ORGANIZATION = 'OCA'
+DEPENDENCIES_FILE = 'oca_dependencies.txt'
+REGEX_ADDONS_PATH = r'^addons_path\s*=\s*'
+
+
+class Repo(object):
+    """
+    oca_dependencies.txt
+
+    For public repo:
+
+    oca-repo -> https://github.com/OCA/oca-repo (default branch)
+    oca-repo 8.0 -> https://github.com/OCA/oca-repo (branch: 8.0)
+    organization/public-repo -> https://github.com/organization/public-repo (default branch)
+    organization/public-repo 8.0 -> https://github.com/organization/public-repo (branch: 8.0)
+    https://github.com/organization/public-repo -> https://github.com/organization/public-repo (default branch)
+    https://github.com/organization/public-repo 8.0 -> https://github.com/organization/public-repo (branch: 8.0)
+    public_repo_rename https://github.com/organization/public-repo -> https://github.com/organization/public-repo (default branch)
+    public_repo_rename https://github.com/organization/public-repo 8.0 -> https://github.com/organization/public-repo (branch: 8.0)
+
+    For private repo:
+
+    git@github.com:Elico-Corp/private-repo
+    git@github.com:Elico-Corp/private-repo 8.0
+    private_repo_rename git@github.com:Elico-Corp/private-repo
+    private_repo_rename git@github.com:Elico-Corp/private-repo 8.0
+
```

Comment:
A killer feature would be to select a list of modules we want to include from a repo. It would avoid:

- cloning the full repo in case we need only a part of it
- have a huge list of modules loaded by Odoo and available in the modules list

For instance, if we want to use only the module `project_earned_value_task` from the branch `8.0-project_earned_value_task` of the repo `https://github.com/Eficent/project`, we could include it like this:

`https://github.com/Eficent/project 8.0-project_earned_value_task project_earned_value_task`

If we want to include the modules `website_blog_rss` and `website_product_rss` from the `https://github.com/Vauxoo/addons-vauxoo` repo, we could use:

`https://github.com/Vauxoo/addons-vauxoo 8.0 website_blog_rss website_product_rss`

Main problem with this notation is that we have to specify the branch systematically. Maybe we could use a named parameter such as `-m` / `--modules` for that matter.

Please add a `TODO` here (simply copy/paste my comment)